Initial UI

keithrpotempa opened the pull request:
### Notes
- This comprises functional MVP; styling to come, but the basic objectives have been hit

### Change Summary
- Added Material UI
- Added Default styled Card component and currency formatting
- Added simple Add/Remove item from state functionality
- Grouping items by type 
- Limiting the item selection to one per type
- Added PriceRange component that calculates and displays the current price range
- Broke out helper component format-currency
- Formatting type names without underscores and properly capitalized
- Budget state and input added
- Price range component calculating whether we are within budget or not

Commit:
Added simple adding and removing item functionality

Default styled Card component and currency formatting

ghost commented:
# [Gatsby Cloud](https://www.gatsbyjs.com) Build Report
**budget-calculator-exercise**

:tada: Your build was successful! [See the Deploy preview here.](https://budgetcalculatorexercise-featureinitialui.gtsb.io)

## Build Details

[View the build logs here.](https://gatsbyjs.com/dashboard/ae512f14-a77a-4f8e-a0e5-3ab0d2960d3f/sites/963973b2-b861-498e-8dde-0437053b4505/builds/c826e401-e895-404a-acd8-e7d03337c6f8/details)

:clock1: Build time: **2m**

## Performance

### Lighthouse report

| Metric | Score |
| ----- | ---- |
| Performance |   :green_heart: 90   |
| Accessibility |   :green_heart: 98   |
| Best Practices |   :green_heart: 100   |
| SEO |   :green_heart: 91   |

:link: [View full report](https://build-c826e401-e895-404a-acd8-e7d03337c6f8.gtsb.io/reports/lighthouse/index.html)

Commit:
Grouping items by type

Created new component item-type-group and organizing items by group to be passed down to it

Commit:
Limiting the item selection to one per type

Commit:
Selected Price range added

Added PriceRange component that calculates and displays the current price range

Broke out helper component format-currency

Formatting type names without underscores and properly capitalized

Commit:
MVP; price range input and calculations

Budget state and input added

Price range component calculating whether we are within budget or not

Commit:
Switched require with import for gatsby-requirements

Commit:
Added gastby-plugin-material-ui and defined ApiManager as const

Commit:
Switched over to gastby-plugin-firebase due to compatability issues
